pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -370,49 +420,50 @@ func (r *ReconcilePolicyRecommendation) Reconcile(ctx context.Context, request r
 	}
 
 	if !isManagedCluster {
-		certificateManager, err := certificatemanager.Create(r.client, network, r.clusterDomain, common.OperatorNamespace())
+		certificateManager, err := certificatemanager.Create(r.client, args.Installation, r.clusterDomain, request.TruthNamespace())
 		if err != nil {
-			r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
+			r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, logc)
 			return reconcile.Result{}, err
 		}
 
 		var managerInternalTLSSecret certificatemanagement.CertificateInterface
 		if managementCluster != nil {
-			managerInternalTLSSecret, err = certificateManager.GetCertificate(r.client, render.ManagerInternalTLSSecretName, common.OperatorNamespace())
+			managerInternalTLSSecret, err = certificateManager.GetCertificate(r.client, render.ManagerInternalTLSSecretName, request.TruthNamespace())
 			if err != nil {
-				r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("failed to retrieve / validate  %s", render.ManagerInternalTLSSecretName), err, reqLogger)
+				r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("failed to retrieve / validate  %s", render.ManagerInternalTLSSecretName), err, logc)
 				return reconcile.Result{}, err
 			}
 		}
 
 		linseedCertLocation := render.TigeraLinseedSecret
-		linseedCertificate, err := certificateManager.GetCertificate(r.client, linseedCertLocation, common.OperatorNamespace())
+		linseedCertificate, err := certificateManager.GetKeyPair(r.client, linseedCertLocation, request.TruthNamespace())
```

Comment:
We need to keep this as `GetCertificate` - this controller shouldn't need the full key pair, it should just need the certificate from the pair.